compiler/rustc_target/src/asm/powerpc.rs

```diff
@@ -44,75 +44,75 @@ impl PowerPCInlineAsmRegClass {
-        f18: freg = ["f18"],
-        f19: freg = ["f19"],
-        f20: freg = ["f20"],
-        f21: freg = ["f21"],
-        f22: freg = ["f22"],
-        f23: freg = ["f23"],
-        f24: freg = ["f24"],
-        f25: freg = ["f25"],
-        f26: freg = ["f26"],
-        f27: freg = ["f27"],
-        f28: freg = ["f28"],
-        f29: freg = ["f29"],
-        f30: freg = ["f30"],
-        f31: freg = ["f31"],
-        #error = ["r1", "sp"] =>
+        r0: reg = ["r0", "0"],
+        r3: reg, reg_nonzero = ["r3", "3"],
+        r4: reg, reg_nonzero = ["r4", "4"],
+        r5: reg, reg_nonzero = ["r5", "5"],
+        r6: reg, reg_nonzero = ["r6", "6"],
+        r7: reg, reg_nonzero = ["r7", "7"],
+        r8: reg, reg_nonzero = ["r8", "8"],
+        r9: reg, reg_nonzero = ["r9", "9"],
+        r10: reg, reg_nonzero = ["r10", "10"],
+        r11: reg, reg_nonzero = ["r11", "11"],
+        r12: reg, reg_nonzero = ["r12", "12"],
+        r14: reg, reg_nonzero = ["r14", "14"],
+        r15: reg, reg_nonzero = ["r15", "15"],
+        r16: reg, reg_nonzero = ["r16", "16"],
+        r17: reg, reg_nonzero = ["r17", "17"],
+        r18: reg, reg_nonzero = ["r18", "18"],
+        r19: reg, reg_nonzero = ["r19", "19"],
+        r20: reg, reg_nonzero = ["r20", "20"],
+        r21: reg, reg_nonzero = ["r21", "21"],
+        r22: reg, reg_nonzero = ["r22", "22"],
+        r23: reg, reg_nonzero = ["r23", "23"],
+        r24: reg, reg_nonzero = ["r24", "24"],
+        r25: reg, reg_nonzero = ["r25", "25"],
+        r26: reg, reg_nonzero = ["r26", "26"],
+        r27: reg, reg_nonzero = ["r27", "27"],
+        r28: reg, reg_nonzero = ["r28", "28"],
+        f0: freg = ["f0" /*, "0" */],
+        f1: freg = ["f1" /*, "1" */],
+        f2: freg = ["f2" /*, "2" */],
+        f3: freg = ["f3" /*, "3" */],
+        f4: freg = ["f4" /*, "4" */],
+        f5: freg = ["f5" /*, "5" */],
+        f6: freg = ["f6" /*, "6" */],
+        f7: freg = ["f7" /*, "7" */],
+        f8: freg = ["f8" /*, "8" */],
+        f9: freg = ["f9" /*, "9" */],
+        f10: freg = ["f10" /*, "10" */],
+        f11: freg = ["f11" /*, "11" */],
+        f12: freg = ["f12" /*, "12" */],
+        f13: freg = ["f13" /*, "13" */],
+        f14: freg = ["f14" /*, "14" */],
+        f15: freg = ["f15" /*, "15" */],
+        f16: freg = ["f16" /*, "16" */],
+        f17: freg = ["f17" /*, "17" */],
+        f18: freg = ["f18" /*, "18" */],
+        f19: freg = ["f19" /*, "19" */],
+        f20: freg = ["f20" /*, "20" */],
+        f21: freg = ["f21" /*, "21" */],
+        f22: freg = ["f22" /*, "22" */],
+        f23: freg = ["f23" /*, "23" */],
+        f24: freg = ["f24" /*, "24" */],
+        f25: freg = ["f25" /*, "25" */],
+        f26: freg = ["f26" /*, "26" */],
+        f27: freg = ["f27" /*, "27" */],
+        f28: freg = ["f28" /*, "28" */],
+        f29: freg = ["f29" /*, "29" */],
+        f30: freg = ["f30" /*, "30" */],
+        f31: freg = ["f31" /*, "31" */],
```

Comment:
You don't need aliases for floating-point registers here. Or if you really want you can add `frN` which clang accepts as an alias for `fN`.